stream_alert/rule_processor/rules_engine.py

```diff
@@ -420,34 +426,36 @@ def process(cls, input_payload):
         return alerts
 
     @classmethod
-    def get_intelligence(cls, intel_dir='threat_intel'):
+    def load_intelligence(cls, intel_dir='threat_intel'):
         """Load intelligence from csv.gz files into a dictionary
 
         Args:
             intel_dir (str): Location where stores compressed intelligence
         """
-        threat_intel = ThreatIntel(intel_dir)
         if cls.__intelligence:
             return
+        threat_intel = StreamThreatIntel(intel_dir)
         cls.__intelligence = threat_intel.read_compressed_files()
 
     @classmethod
-    def is_ioc(cls, rec):
-        """Detect if a record contains any data matching to IOC (Indicate of
-        compromise) which store in class variable intelligence dictionary.
+    def is_ioc(cls, rec, datatypes_ioc_mapping=None):
+        """Detect if a record contains any data matching to known IOCs
+
+        A class variable is defined to store IOCs (Indicator of Compromise) in a
+        dictionary.
 
         Args:
             rec (dict): Payload record to process
 
         Return:
             (tuple): (True or False, IOC type, IOC value)
         """
-        _, datatypes_ioc_mapping = load_threat_intel_conf()
-        for datatype in rec['normalized_types']:
-            if datatype not in datatypes_ioc_mapping:
-                continue
-            results = fetch_values_by_datatype(rec, datatype)
-            for result in results:
-                if result in cls.__intelligence[datatypes_ioc_mapping[datatype]]:
-                    return True, datatypes_ioc_mapping[datatype], result
+        if datatypes_ioc_mapping:
```

Comment:
suggestion here to avoid multi-level nesting so much would be to return early if possible. ie:
```
if not datatypes_ioc_mapping:
    return False, None, None
```